src/libLookup/Lookup.cpp

```diff
@@ -1590,6 +1590,99 @@ void Lookup::SendGetMicroBlockFromLookup(const vector<BlockHash>& mbHashes) {
   SendMessageToRandomLookupNode(msg);
 }
 
+bool Lookup::ProcessGetCosigsRewardsFromSeed(
+    [[gnu::unused]] const bytes& message, [[gnu::unused]] unsigned int offset,
+    [[gnu::unused]] const Peer& from) {
+  LOG_MARKER();
+
+  if (!LOOKUP_NODE_MODE) {
+    LOG_GENERAL(WARNING,
+                "Function not expected to be called from non-lookup node");
+    return false;
+  }
+
+  // verify if sender is from know DS Committee
+  uint128_t ipSenderAddr = from.m_ipAddress;
+  bool isDSNode = false;
+  for (const auto& node : *m_mediator.m_DSCommittee) {
```

Comment:
Since the complexity is slow here. Is it possible to make DSCommittee a set instead of deque in another PR for m_DSCommittee?

Reply:
yeah. can consider. will make a note,